src/pages/index.js

```diff
@@ -107,32 +134,83 @@ function closeModalByEscape(event) {
 // Handle Edit Form Submit
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = profileNameInput.value;
-  profileDescription.textContent = profileDescriptionInput.value;
-  closeModal(editModal);
+  const saveButton = editForm.querySelector(".modal__submit-btn");
+  saveButton.textContent = "Saving...";
+
+  api
+    .editUserInfo({
+      name: profileNameInput.value,
+      about: profileDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      saveButton.textContent = "Save";
+      closeModal(editModal);
+    })
+    .catch((err) => {
+      saveButton.textContent = "Save";
+      console.error(err);
+    });
 }
 
 // Handle Add Card Submit
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
-  const newCard = getCardElement({
+
+  const saveButton = cardForm.querySelector(".modal__submit-btn");
+  saveButton.textContent = "Saving...";
+
+  const cardData = {
     name: cardNameInput.value,
     link: cardLinkInput.value,
-  });
-  cardsList.prepend(newCard);
-  closeModal(cardModal);
-  cardForm.reset();
+  };
 
-  toggleButtonState(
-    Array.from(cardForm.querySelectorAll(validationConfig.inputSelector)),
-    cardForm.querySelector(validationConfig.submitButtonSelector),
-    validationConfig
-  );
+  api
+    .addCard(cardData)
+    .then((newCard) => {
+      const cardElement = getCardElement(newCard);
+      cardsList.prepend(cardElement);
+      cardForm.reset();
+      saveButton.textContent = "Save";
+      closeModal(cardModal);
+
+      toggleButtonState(
+        Array.from(cardForm.querySelectorAll(validationConfig.inputSelector)),
+        cardForm.querySelector(validationConfig.submitButtonSelector),
+        validationConfig
+      );
+    })
+    .catch((err) => {
+      saveButton.textContent = "Save";
+      console.error(err);
+    });
+}
+
+//Avatar Handler
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const saveButton = avatarForm.querySelector(".modal__submit-btn");
+  saveButton.textContent = "Saving...";
+  saveButton.disabled = true;
```

Comment:
```js

  saveButton.disabled = true;
  saveButton.classList.add("modal__submit-btn_disabled");
```

Please, reuse `toggleButtonState` not to duplicate the same code of disabling

You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.